app/js/app-menu.js

```diff
@@ -38,42 +38,74 @@ module.exports = function () {
     return chise.loadSample(filename, 'app/samples/');
   }
 
-  function loadFromURL(){
-    var filePath = getParameterByName('filename');
-
+  function loadFromURL(filepath){
     var loadCallbackSBGNMLValidity = function (text) {
       validateSBGNML(text);
     }      
     var loadCallbackInvalidityWarning  = function () {
       promptInvalidFileView.render();
     }
 
-    if(filePath == undefined)
+    if(filepath == undefined)
       return;
 
-    if(filePath.indexOf('file:///') === 0){
-      console.log('read from local');
-    }
-    else if(filePath.indexOf('http') === 0){
-      //load file via url from remote
+    //load file via url from remote
+    if(filepath.indexOf('http') === 0){
+      var filename = filepath.split('/');
+      if(filename.length > 0)
+        filename = filename[filename.length - 1];
+      else
+        filename = 'remote';
+
+      var fileExtension = filename.split('.');
+      if(fileExtension.length > 0)
+        fileExtension = fileExtension[fileExtension.length - 1];
+      else
+        fileExtension = 'txt';
+
       $.ajax({
-        url: filePath,
+        type: 'get',
+        url: filepath,
         success: function(result){
-          var fileToLoad = new File([result], 'remote-file.xml', {
-            type: 'text/xml',
+          var fileToLoad = new File([result], filename, {
+            type: 'text/' + fileExtension,
             lastModified: Date.now()
           });
+          
+          chise.loadSBGNMLFile(fileToLoad, loadCallbackSBGNMLValidity, loadCallbackInvalidityWarning);
+        },
+        error: function(xhr, ajaxOptions, thrownError){
+          promptInvalidFileView.render();
+          $("#new-file").trigger('click');
+        }
+      });       
+    }
+  }
 
-          if(cy.elements().length != 0) {
-            promptConfirmationView.render(function(){chise.loadSBGNMLFile(fileToLoad, loadCallbackSBGNMLValidity, loadCallbackInvalidityWarning)});
+  function loadFromURI(uri){
+
+    var queryURL = "http://www.pathwaycommons.org/pc2/get?uri="
+          + uri + "&format=SBGN";
+
+    var filename = uri + '.sbgnml';
+    chise.startSpinner('paths-byURI-spinner');
+
+    $.ajax({
+        url: queryURL,
+        type: 'GET',
+        success: function (data) {
+          if (data == null) {
```

Comment:
@ahmetcandiroglu also warn the user about invalid URI, similar to this: https://github.com/iVis-at-Bilkent/newt/blob/2c7697db7e613f43dcb5de94003a1573c35b3760/app/js/backbone-views.js#L790